frontend/src/components/App/icons.test.ts

```diff
@@ -0,0 +1,40 @@
+import { error } from 'console';
+import fs from 'fs';
+import glob from 'glob';
+import mdiIcons from './icons';
+
+// the usedIcons array is used to check that all icons are used in the frontend
+// we take the keys from the mdiIcons and add the prefix to it as a string
+const usedIcons = Object.keys(mdiIcons.icons).map(icon => `${mdiIcons.prefix}:${icon}`);
+
+const checkIcons = async () => {
+  const files = glob.sync('./**/*.tsx', { ignore: './node_modules/**' });
+  for (const file of files) {
+    const content = await fs.readFileSync(file, 'utf8');
+    // const regex = /mdi:\w+/g;
+    // \w+ matches any word character (equal to [a-zA-Z0-9_])
+    // + Quantifier — Matches between one and unlimited times, as many times as possible, giving back as needed (greedy)
+    // g modifier: global. All matches (don't return after first match)
+    // this will only read for words, not for words with a dash in it
+
+    // this regex will match all mdi:icon-name
+    const regex = /mdi:[\w-]+/g;
+    let match;
+    while ((match = regex.exec(content)) !== null) {
+      if (usedIcons.includes(match[0])) {
+        return true;
+      } else {
+        // console.log(`Icon ${match[0]} is not used in ${file}`);
+        error(`Icon ${match[0]} is not used in ${file}`);
+        return false;
+      }
+    }
+  }
+};
+
+describe('Icon tests', () => {
+  test('Check icons', async () => {
+    const result = await checkIcons();
+    expect(result).toBe(true);
```

Comment:
The logic for testing is correct. This is exactly what we talked about.
It can be improved though, to make the tests more understandable when they fail, by having the function for checking icons create a test (`test('Check icon ${match[0]}', async () => { expect(usedIcons.includes(matches[0])).toBe(true) }`) for each icon that the regex matches. So when the test fails, we see exactly which icon failed.

Reply:
It seems to print the failed icon near the test failure...

From the test failure:
> Icon mdi:more-vert is not used in ./src/components/App/Home/index.tsx

But maybe that message could be improved by telling people what they need do?